modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/macros/CP2ActivationKeys.macro

```diff
@@ -302,6 +302,35 @@ definition {
 		return "${downloadedFileName}";
 	}
 
+	macro generateKeyFlow {
+		CustomerPortal.search(searchItem = "NASA ICE");
+
+		CustomerPortal.enterOnProject(title = "NASA ICE");
+	
+		CP2NavigationMenu.goToProductActivation();
+
+		CP2NavigationMenu.clickOnProductButton(product = "Portal");
+		
+		CP2ActivationKeys.generateNewKey();
+
+		CP2ActivationKeys.selectGenerateActivationKeyOptionByField(
+			field = "Product",
+			option = "Portal");
+
+		CP2ActivationKeys.selectGenerateActivationKeyOptionByField(
+			field = "Version",
+			option = "6.1 GA1");
+
+		CP2ActivationKeys.selectGenerateActivationKeyOptionByField(
+			field = "Key Type",
+			option = "Production");
+
+		CP2ActivationKeys.selectFistActiveSubscriptionInput();
+
+		CustomerPortal.clickAtButton(text = "Next");
+	
+	}
+
```

Comment:
Hi @luanakarla, I think that you can put variables on this macro instead of the values. If you do this, the macro will be more useful for another options that go beyond 'NASA ICE', for example.